packages/web-components/src/components/expressive-modal/expressive-modal.scss

```diff
@@ -17,6 +17,10 @@
     @extend :host(#{$prefix}-modal[open]);
   }
 
+  &:not([open]) {
+    display: none;
+  }
```

Comment:
Please make sure working with designers. IIRC modals in Carbon have fading animations for closing and `display:none` for hidden modal would kill such animation.